atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/FileBasedReceivedSnapshot.java

```diff
@@ -192,27 +192,29 @@ public PersistedSnapshot persist() {
     final var files = directory.toFile().listFiles();
     Objects.requireNonNull(files, "No chunks have been applied yet");
 
-    if (files.length != expectedTotalCount) {
+    if (expectedTotalCount != Integer.MIN_VALUE && files.length != expectedTotalCount) {
```

Comment:
I think it feels less hacky if you name the constants. Like FALLBACK_TOTAL_COUNT or something